src/features/core/cookieConsent/ketch/browser.js

```diff
@@ -0,0 +1,112 @@
+import logger from '../../../../utils/logUtil';
+import { Cookie } from '../../../../utils/storage/cookie';
+
+/* eslint-disable class-methods-use-this */
+class Ketch {
+  constructor() {
+    this.userConsentedPurposes = [];
+    this.userDeniedPurposes = [];
+
+    // updateKetchConsent callback function to update current consent purpose state
+    // this will be called from ketch rudderstack plugin
+    window.updateKetchConsent = (consent) => {
+      if (consent) {
+        this.userConsentedPurposes = [];
+        this.userDeniedPurposes = [];
+        this.updatePurposes(consent);
+      }
+    };
+
+    // getKetchUserConsentedPurposes returns current ketch opted-in purposes
+    window.getKetchUserConsentedPurposes = () => this.userConsentedPurposes.slice();
+
+    // getKetchUserDeniedPurposes returns current ketch opted-out purposes
+    window.getKetchUserDeniedPurposes = () => this.userDeniedPurposes.slice();
+
+    // initialize purposes with existing user consent state
+    if (window.ketchConsent) {
+      this.updatePurposes(window.ketchConsent);
+    } else {
+      const consent = this.getConsent();
+      this.updatePurposes(consent);
+    }
+  }
+
+  updatePurposes = (consent) => {
+    Object.entries(consent).forEach((e) => {
+      const purposeCode = e[0];
+      const isConsented = e[1];
+      if (isConsented) {
+        this.userConsentedPurposes.push(purposeCode);
+      } else {
+        this.userDeniedPurposes.push(purposeCode);
+      }
+    });
```

Comment:
Add a checking here, if consent is available then only loop through the consent.